### PR TITLE
Make cookbook/benchmark validation more robust

### PR DESCRIPTION
**Pull request by gassmoeller**
#5259 is failing because the cookbook/benchmark validation is not very robust concerning different path layouts. Make the test more robust:
- automatically build all plugins in all subdirectories of the given cookbook/benchmark
- by default try to run all .prm files in all subdirectories (except for doc/)
- remove all special rules that are no longer necessary

@tjhei: can you take a look?

### CODE REVIEW

File: benchmarks/check.mk [hunk 1]
```diff
@@ -74,33 +79,6 @@ main: dummy
...
-	@$(def); run_all_prms $@
-
-crameri_et_al/:  dummy
-	+@$(def); make_lib $@/case_1
-	@$(def); run_all_prms $@/case_1
-	@$(def); run_all_prms $@/case_2
-
-davies_et_al/: dummy
-	+@$(def); make_lib $@/case-2.3-plugin
-	@$(def); run_prm $@ case-2.1.prm
```
Comment (tjhei) [inline benchmarks/check.mk:93]: Are you sure this works? I vaguely recall that the other prm files resume this file.

**Review by tjhei — APPROVED** [review]
I am ok with building all plugins, no harm here.

I am not sure your other changes will work: in CI we only validate and not run all examples. For running, many special cases are necessary (files use resume features, others don't converge, etc).

**Comment by gassmoeller** [comment]
Hmm, I guess you are right, I didnt try running the benchmarks. Let me think about this, maybe I split the PR up into the part dealing with plugins and the part dealing with prms.

**Comment by gassmoeller** [comment]
I think this will get too complicated to be worth the effort. Closing for now, we can revisit if necessary.

**Pull request closed without merging**